queries/highlights.scm

```diff
@@ -162,15 +168,244 @@
   "isnot"
   "cast"
 ] @keyword.operator
+; operators that are English words (e.g. `and` / `or`)
 
 [
   "return"
   "yield"
+  "discard"
 ] @keyword.return
+; keywords like `return` and `yield`
 
-; Operators
-(infix_expression operator: _ @operator)
-(prefix_expression operator: _ @operator)
 [
-  "="
-] @operator
+  "if"
+  "when"
+  "case"
+  "elif"
+  "else"
+] @conditional
+(of_branch "of" @conditional)
+; keywords related to conditionals (e.g. `if` / `else`)
+
+[
+  "for"
+  "while"
+  "continue"
+  "break"
+] @repeat
+(for "in" @repeat)
+; keywords related to loops (e.g. `for` / `while`)
+
+[
+  "block"
+] @label
+; GOTO and other labels (e.g. `label:` in C)
```

Comment:
This is definitely not a label.

If anything, the label is what follows `block`.